applications/StructuralMechanicsApplication/custom_constitutive/truss_plasticity_constitutive_law.h

```diff
@@ -212,21 +236,21 @@ class KRATOS_API(STRUCTURAL_MECHANICS_APPLICATION) TrussPlasticityConstitutiveLa
     {
         KRATOS_SERIALIZE_SAVE_BASE_CLASS( rSerializer, ConstitutiveLaw);
         rSerializer.save("StressState", this->mStressState);
```

Comment:
hm I am not sure if this might become a problem in terms of optimization
@armingeiser this is probably sth to discuss

Reply:
It will definitely lead to problems with adjoint sensitivity analysis, but it was already there, so I don't know if we should/can solve it right in this PR unless we have a proper usecase.

Reply:
It was added very recently so now might still be the chance to find a better solution

@KlausBSautter I am approving (consider my other comments pls though)
Still this is sth we should speak abt, otherwise we will end up with the same problem that we have in the shells ...

note to myself: create PerturbationTests!